primitives/state-machine/src/ext.rs

```diff
@@ -346,45 +384,79 @@ where
 		}
 	}
 
-	fn next_child_storage_key(&self, child_info: &ChildInfo, key: &[u8]) -> Option<StorageKey> {
-		let mut next_backend_key = self
-			.backend
-			.next_child_storage_key(child_info, key)
-			.expect(EXT_NOT_ALLOWED_TO_FAIL);
-		let mut overlay_changes =
-			self.overlay.child_iter_after(child_info.storage_key(), key).peekable();
+	fn next_child_storage_key(
+		&mut self,
+		child_info: &ChildInfo,
+		key: &[u8],
+		count: u32,
+	) -> Option<Vec<StorageKey>> {
```

Comment:
Note this rewrite is to return a Vec of the next keys, it is not necessary for default child trie but is for the transient storage: we can be conservative and keep old implemetation for default child trie (but would make sense to switch is api to a vec of next keys too).